solutions/usaco-1181.mdx

```diff
@@ -0,0 +1,97 @@
+---
+id: usaco-1181
+source: USACO Bronze 2022 January
+title: Drought
+author: Chongtian Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_prob3_bronze_jan22.html)
+
+## Explanation
+
+We can solve this problem greedily using one linear sweep and making prefixes of cows have equal hunger as we sweep from left to right. Thus, we can break this
+problem up into cases:
+
+Suppose the cows are numbered from $1$ to $n$, and we are currently looking at cow $i$. If cow $i+1$ has a greater hunger value, then it is sufficient to decrease cow $i+1$ to the same hunger valueas cow $i$. To do this we will pick the pair of cows $i+1$ and $i+2$ and decrease their hungers until cow $i+1$ has the same hunger value as cow $i$. Of course, if cow $i+2$ has a negative hunger value, then it is impossible and we output $-1$.
```

Comment:
1. what does "sufficient" mean?
2. wdym by "negative hunger value"? aren't the given hunger values positive?

Reply:
1. what doesn't make sense here
2. read the problem

Reply:
1. sufficient means just enough
2. you dont want to subtract the hunger into negative. i changed the wording here alittle.

Reply:
> 1. what doesn't make sense here
> 2. read the problem

not helpful jesse